api/src/test/java/org/apache/iceberg/TestHelpers.java

```diff
@@ -54,7 +54,7 @@ public class TestHelpers {
 
   private TestHelpers() {}
 
-  public static final int MAX_FORMAT_VERSION = 3;
+  public static final int MAX_FORMAT_VERSION = 4;
```

Comment:
Since v4 spec is not there, I am not sure we should enable this in code at the moment. 

What if user configures v4, what does it actually mean a v4 table?

Reply:
We need this to enable v4 Dev work

Reply:
we did the same with #10760 which was added right around the same time when v3 Spec work happened. Since there will be multiple efforts working on v4 stuff the v4 base classes should be added to `main` to make development easier for everyone

Reply:
We can add a line to the Spec.md that V3 is done and V4 is now open and experimental

Reply:
Maybe start with one of the v4 feature spec. Get it approved and add this along with the feature.

Reply:
By the time this code is released to users we should have v4 Spec items documented. As I mentioned earlier, we did the same approach with v3 as well

Reply:
I think it’s worth focusing on what’s technically correct rather than solely relying on what has been done previously. Since we publish nightly builds, there’s a chance users could start creating v4 tables before any spec is officially finalized. With the 1.10.0 discussions already underway, this change could potentially be included in a release without any approved v4 spec.

My suggestion was simply to finalize at least one v4 spec change before beginning v4 development on the main branch, so that users have a clear understanding of what features come with enabling v4. 

That said, I may be overthinking this or being too cautious 😅. Please feel free to go with the broader consensus from the community on this PR (other reviewers opinion).

Reply:
Maybe we can update `SUPPORTED_TABLE_FORMAT_VERSION` later.

Reply:
@manuzhang yes, otherwise you won't be able to create a v4 table for testing

Reply:
I've talked offline with @manuzhang and he's fine with this since we already did the same thing for v3